internal/service/cloud/network.go

```diff
@@ -162,6 +163,11 @@ func (s *Service) getLAN(ctx context.Context, ms *scope.Machine) (*sdk.Lan, erro
 	)
 
 	for _, l := range *lans.Items {
+		// Use an existing LAN if it's specified in the spec.
+		if ms.IonosMachine.Spec.LanID != nil && ptr.Equal(l.GetId(), ms.IonosMachine.Spec.LanID) {
+			return &l, nil
+		}
```

Comment:
Comparing with the `LanProperties` we'd set in `createLAN`, I wonder what would happen if the CR contains a LAN ID of a private LAN. Does it break somewhere with a meaningful error? Or do we need validation and documentation for this requirement?
Similar thing for the `Ipv6CidrBlock` field: We explicitly enable IPv6 by setting it to `AUTO`. In user-provided LAN this might not be the case, leading to issues later on.

Reply:
I'm ok with making this the user's responsibility instead of adding all kinds of validation now. But the user needs to know at least what is right and what is wrong, so these requirements need to be documented. I think it should be enough if the godocs for the new `NetworkID` say that the LAN need to be a public one and that IPv6 must be enabled.